tests[DI-25148]grouping the alert for show deatils

agorthi-akamai opened the pull request:
## Description 📝

Highlight the Pull Request's context and intentions.

## Changes  🔄

List any change(s) relevant to the reviewer.

- ...
- ...

## Target release date 🗓️

Please specify a release date (and environment, if applicable) to guarantee timely review of this PR. If exact date is not known, please approximate and update it as needed.

## Preview 📷

**Include a screenshot or screen recording of the change.**

:lock: Use the [Mask Sensitive Data](https://cloud.linode.com/profile/settings) setting for security.

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
| 📷 | 📷 |

## How to test 🧪

### Prerequisites

(How to setup test environment)

- ...
- ...

### Reproduction steps

(How to reproduce the issue, if applicable)

- [ ] ...
- [ ] ...

### Verification steps

(How to verify changes)

- [ ] ...
- [ ] ...

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

*Check all that apply*

- [ ] Use React components instead of HTML Tags
- [ ] Proper naming conventions like cameCase for variables & Function & snake_case for constants
- [ ] Use appropriate types & avoid using "any"
- [ ] No type casting & non-null assertions
- [ ] Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] Providing/Improving test coverage
- [ ] Use sx props to pass styles instead of style prop
- [ ] Add JSDoc comments for interface properties & functions
- [ ] Use strict equality (===) instead of double equal (==)
- [ ] Use of named arguments (interfaces) if function argument list exceeds size 2
- [ ] Destructure the props
- [ ] Keep component size small & move big computing functions to separate utility
- [ ] 📱 Providing mobile support

<br/>

- [ ] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [ ] All unit tests are passing
- [ ] TypeScript compilation succeeded without errors
- [ ] Code passes all linting rules

</details>

---

## Commit message and pull request title format standards

> **Note**: Remove this section before opening the pull request
**Make sure your PR title and commit message on squash and merge are as shown below**

`<commit type>: [JIRA-ticket-number] - <description>`

**Commit Types:**

- `feat`: New feature for the user (not a part of the code, or ci, ...).
- `fix`: Bugfix for the user (not a fix to build something, ...).
- `change`: Modifying an existing visual UI instance. Such as a component or a feature.
- `refactor`: Restructuring existing code without changing its external behavior or visual UI. Typically to improve readability, maintainability, and performance.
- `test`: New tests or changes to existing tests. Does not change the production code.
- `upcoming`: A new feature that is in progress, not visible to users yet, and usually behind a feature flag.

**Example:** `feat: [M3-1234] - Allow user to view their login history`

---

venkymano-akamai commented on packages/manager/cypress/e2e/core/cloudpulse/alert-show-details.spec.ts:
```diff
@@ -167,242 +158,315 @@ describe('Integration Tests for Alert Show Detail Page', () => {
+        updated_by: 'user2',
+        created: '2023-10-01T12:00:00Z',
+        updated: new Date().toISOString(),
+        group: group as RecPartial<AlertDefinitionGroup>,
```
why can't we avoid this type casting

venkymano-akamai commented on packages/manager/cypress/e2e/core/cloudpulse/alert-show-details.spec.ts:
```diff
@@ -167,242 +158,315 @@ describe('Integration Tests for Alert Show Detail Page', () => {
-      cy.findByText('To:').should('be.visible');
-      cy.findByText('test@test.com').should('be.visible');
-      cy.findByText('test2@test.com').should('be.visible');
+      scopeActions[scopeLabel]?.();
```
??

nikhagra-akamai commented on packages/manager/cypress/e2e/core/cloudpulse/alert-show-details.spec.ts:
```diff
@@ -167,242 +158,315 @@ describe('Integration Tests for Alert Show Detail Page', () => {
+    },
+    Account: () => {
+      const expectedMessage =
+        'All entities associated with current account will be included in this alert definition. Any new entity created with this account will also be included.';
```
I've created a  constant for this in this file

https://github.com/ACLPManager/manager/blob/cb4d49e32db729399bd418b8c9e014bf1762bdff/packages/manager/src/features/CloudPulse/Alerts/constants.ts#L225

nikhagra-akamai commented on packages/manager/cypress/e2e/core/cloudpulse/alert-show-details.spec.ts:
```diff
@@ -167,242 +158,315 @@ describe('Integration Tests for Alert Show Detail Page', () => {
+    },
+  };
+
+  const scopes = [
+    { group: 'per-region', scopeLabel: 'Region' },
+    { group: 'per-entity', scopeLabel: 'Entity' },
+    { group: 'per-account', scopeLabel: 'Account' },
+  ];
```
You can use any of these constants instead of redefining it again or typecasting

https://github.com/ACLPManager/manager/blob/cb4d49e32db729399bd418b8c9e014bf1762bdff/packages/manager/src/features/CloudPulse/Alerts/constants.ts#L109

https://github.com/ACLPManager/manager/blob/cb4d49e32db729399bd418b8c9e014bf1762bdff/packages/manager/src/features/CloudPulse/Alerts/constants.ts#L115

agorthi-akamai commented:
I have addressed the review comments. Please review the changes and provide your feedback

venkymano-akamai reviewed (approved):
@agorthi-akamai , please see if we can fix the ESLint warning

ankita-akamai commented:
@agorthi-akamai check the failing cypress ones and above comment regarding eslint fixes.

agorthi-akamai commented:
I have fixed one ESLint issue on the Show Details page. However, I couldn’t resolve the other ESLint issue related to duplicate strings.
This is causing failures in the Cypress create alert tests, specifically due to the Scope/Grouping feature.
I have already addressed this issue in https://github.com/ACLPManager/manager/pull/327.

Please review and approve my current PR. Thank you!
